server/chat-plugins/room-faqs.ts

```diff
@@ -133,6 +140,46 @@ export const commands: ChatCommands = {
 	],
 };
 
+export const pages: PageTable = {
+	roomfaqs(args, user) {
+		const room = this.requireRoom();
+		this.title = `[Room FAQs]`;
+		// allow it for users if they can access the room
+		if (!user.inRooms.has(room.roomid) && room.settings.isPrivate && !user.isStaff) {
+			return this.errorReply(`Access denied.`);
+		}
+		let buf = `<div class="pad"><button style="float:right;" class="button" name="send" value="/join view-roomfaqs-${room.roomid}"><i class="fa fa-refresh"></i> Refresh</button>`;
+		if (!roomFaqs[room.roomid]) {
+			return `${buf}<h2>This room has no FAQs.</h2></div>`;
+		}
+
+		buf += `<h2>FAQs for ${room.title}:</h2>`;
+		const keys = Object.keys(roomFaqs[room.roomid]).filter(
+			val => !getAlias(room.roomid, val)
+		).sort((a, b) => a.localeCompare(b));
+		for (const key of keys) {
+			const topic = roomFaqs[room.roomid][key];
+			buf += `<div class="infobox">`;
+			buf += `<h3>${key}</h3>`;
+			buf += `<hr />`;
+			buf += Chat.formatText(topic, true).replace(/\n/g, '<br />');
+			const aliases = Object.keys(roomFaqs[room.roomid]).filter(val => getAlias(room.roomid, val) === key);
```

Comment:
```suggestion
		buf += `<h2>FAQs for ${room.title}:</h2>`;
		const roomKeys = Object.keys(roomFaqs[room.roomid]);
		const keys = roomKeys.filter(val => !getAlias(room.roomid, val)).sort((a, b) => a.localeCompare(b));
		for (const key of keys) {
			const topic = roomFaqs[room.roomid][key];
			buf += `<div class="infobox">`;
			buf += `<h3>${key}</h3>`;
			buf += `<hr />`;
			buf += Chat.formatText(topic, true).replace(/\n/g, '<br />');
			const aliases = roomKeys.filter(val => getAlias(room.roomid, val) === key);
```

Reply:
What does this even do except repeat aliases as dupes alongside listing them in the rfaqs themselves?

Reply:
It... doesn't do that? It just simplifies two `Object.keys(roomFaqs[room.roomid])` calls into one constant, and also simplifies some formatting.